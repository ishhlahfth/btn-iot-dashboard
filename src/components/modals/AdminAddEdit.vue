<template>
  <help-modal v-model="modal.deletePhoto" permanent>
    <confirmation
      :title="`Delete Confirmation`"
      :message="`This action cannot be undone, and the picture will permanently deleted, are you sure want to delete ?`"
      @close="modal.deletePhoto = false"
      @cancel="modal.deletePhoto = false"
      @confirm="deletePhoto"
      :confirmLoading="loadingDelete"
      labelOk="Delete"
      labelCancel="Cancel"
    />
  </help-modal>
  <div class="modal-md inner-modal-fixed overflow-auto grid gap-6 p-1">
    <div class="flex justify-between items-center">
      <p class="text-heading4 font-semibold">
        {{ adminType === 'edit' ? 'Edit a new admin' : 'Add a new admin' }}
      </p>
      <help-button
        icon-only
        icon="close"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
    </div>
    <form class="grid gap-4" @submit.prevent="submit">
      <div class="grid auto-rows-max gap-2">
        <help-input label="Name" type="text" placeholder="Input admin's name" v-model="form.name" />
        <p class="text-xsmall text-flame font-medium" v-if="!form.name">
          Your name is required
        </p>
      </div>
      <div class="grid auto-rows-max gap-2">
        <help-input
          label="Email"
          type="email"
          placeholder="Input admin's email"
          v-model="form.email"
        />
        <p class="text-xsmall text-flame font-medium" v-if="!form.email">
          Your email is required
        </p>
      </div>
      <div class="grid auto-rows-max gap-2" v-if="adminType === 'add'">
        <help-input
          v-model="form.password"
          label="Password"
          :type="visiblePassword ? 'text' : 'password'"
          placeholder="Input admin's password"
        >
          <help-button
            type="button"
            icon-only
            :icon="visiblePassword ? 'eye-off' : 'eye'"
            bg-color="transparent"
            color="grey-3"
            @click="visiblePassword = !visiblePassword"
          />
        </help-input>
        <p class="text-xsmall text-flame font-medium" v-if="!form.password">
          Your password is required
        </p>
      </div>
      <div class="grid auto-rows-max gap-2">
        <help-input
          label="Phone Number"
          type="number"
          placeholder="Input admin's phone number"
          v-model="form.phone_number"
        />
        <p class="text-xsmall text-flame font-medium" v-if="!form.phone_number">
          Your phone number is required
        </p>
      </div>
      <div class="grid auto-rows-max gap-2">
        <help-input
          label="Address"
          type="textarea"
          placeholder="Input admin's address"
          v-model="form.address"
        />
        <p class="text-xsmall text-flame font-medium" v-if="!form.address">
          Your address is required
        </p>
      </div>
      <div class="grid auto-rows-max gap-2">
        <div>
          <label class="font-medium">Date of Birth</label>
          <flat-pickr
            v-model="form.date_of_birth"
            :config="config"
            class="form-control my-1 rounded-md w-full p-2.5 border text-md cursor-pointer"
            placeholder="Select Date.."
            name="dateBirth"
          />
        </div>
        <p class="text-xsmall text-flame font-medium" v-if="!form.date_of_birth">role
          Your date birth is required
        </p>
      </div>
      <div>
        <input
          type="file"
          accept="image/*"
          class="hidden"
          ref="bannerImageInput"
          @input="handleChangeImg"
        />
        <help-input
          v-if="adminType === 'add'"
          label="Profile Picture"
          type="file"
          placeholder="Input admin's picture"
          v-model="form.photo"
        />
        <div v-else>
          <label class="font-medium">Profile Picture</label>
          <help-thumbnail class="my-1" width="100%" height="100%" :src="src" />
          <div
            v-if="adminType === 'edit'"
            class="grid grid-flow-col auto-cols-max gap-2 justify-end"
          >
            <span
              class="text-royal cursor-pointer font-medium"
              @click="$refs.bannerImageInput.click()"
            >
              Edit
            </span>
            <span v-if="adminType !== 'edit'" class="text-flame cursor-pointer font-medium" @click="src = ''">Remove</span>
            <span v-if="adminType === 'edit'" class="text-flame cursor-pointer font-medium" @click="modal.deletePhoto = true">Remove</span>
          </div>
        </div>
      </div>
      <div class="grid auto-rows-max gap-2">
        <help-select
          v-model="selectedStatus"
          label="Role"
          :options="roles"
          :position="screenWidth < 640 ? ['top', 'right'] : ['bottom', 'right']"
        />
        <p class="text-xsmall text-flame font-medium" v-if="!selectedStatus">
          Your role is required
        </p>
      </div>
      <div class="grid grid-flow-col gap-2 auto-cols-max justify-end py-2">
        <help-button
          type="button"
          label="cancel"
          bg-color="transparent"
          color="grey-1"
          @click="$emit('close')"
        />
        <help-button
          :label="adminType === 'edit' ? 'save changes' : 'add'"
          :loading="loading"
          :loading-label="adminType === 'edit' ? 'saving' : 'adding'"
        />
      </div>
    </form>
  </div>
</template>

<script>
import { uuid } from 'uuidv4';
import { useToast } from 'vue-toastification';
import { CognitoIdentityClient } from '@aws-sdk/client-cognito-identity';
import { fromCognitoIdentityPool } from '@aws-sdk/credential-provider-cognito-identity';
import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
import HelpInput from '@/components/atoms/Input.vue';
import HelpButton from '@/components/atoms/Button.vue';
import HelpThumbnail from '@/components/atoms/Thumbnail.vue';
import HelpSelect from '@/components/molecules/Select.vue';
import HelpModal from '@/components/templates/Modal.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
import API from '../../apis';

export default {
  name: 'AdminAddEdit',
  props: {
    adminType: {
      type: String,
      default: 'add',
    },
    roles: {
      type: Array,
    },
  },
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      loading: false,
      selectedStatus: { value: '', label: 'Select Admin Role' },
      S3BaseURL: process.env.VUE_APP_S3_BASE_URL,
      imageFile: null,
      hyperlink: '',
      src: '',
      imageIsChanged: false,
      form: {
        name: '',
        email: '',
        phone_number: '',
        address: '',
        role_id: '',
        photo: {},
        password: '',
        date_of_birth: '',
      },
      config: {
        wrap: true, // set wrap to true only when using 'input-group'
        // altFormat: 'y M D',
        // altInput: true,
        // dateFormat: 'y-m-d h:m:s',
        // locale: 'ID', // locale for this instance only,
        enableTime: false,
        enableSeconds: false,
        disableMobile: 'true',
      },
      visiblePassword: false,
      modal: {
        deletePhoto: false,
      },
      loadingDelete: false,
      bannerID: null,
    };
  },
  components: {
    HelpInput,
    HelpButton,
    HelpSelect,
    HelpModal,
    HelpThumbnail,
    Confirmation,
  },

  methods: {
    handleChangeImg(e) {
      if (e.target.files.length) {
        const file = e.target.files[0];
        const fileName = `${uuid()}.${e.target.files[0].type.split('/')[1]}`;
        const url = `${this.S3BaseURL}/${fileName}`;
        this.src = URL.createObjectURL(file);
        this.imageFile = { file, fileName, url };
        this.imageIsChanged = true;
      }
    },
    async deletePhoto() {
      this.loadingDelete = true;
      try {
        await API.delete(`banners/${this.bannerID}`);
        this.toast.success('Product images successfully deleted !');
        this.bannerID = null;
      } catch (error) {
        this.toast.error(error.response?.data?.meta?.message);
      }
      this.src = '';
      this.modal.deletePhoto = false;
      this.loadingDelete = false;
    },
    async submit() {
      // if(!this.form.name)
      if (this.form.photo.length) {
        const file = this.form.photo[0];
        const fileName = `${uuid()}.${this.form.photo[0].type.split('/')[1]}`;
        const url = `${this.S3BaseURL}/${fileName}`;
        this.hyperlink = url;
        this.imageFile = { file, fileName, url };
      }
      if (this.adminType === 'edit') {
        this.form.role_id = this.selectedStatus.value || this.form.role_id;
        const dataToSend = {
          name: this.form.name,
          address: this.form.address,
          email: this.form.email,
          role_id: this.form.role_id,
          is_active: 'TRUE',
          phone_number: this.form.phone_number,
          date_of_birth: this.form.date_of_birth,
        };
        try {
          const {
            data: { data },
          } = await API.patch(`/employees/${this.$store.state.adminDetail.id}`, dataToSend);
          if (this.bannerID) {
            this.handleEditAdmin(data.id);
          } else {
            this.handleAddAdmin(data.id, `${this.form.name}'s profile has been successfully edited !`);
          }
        } catch (error) {
          if (error.response?.data?.meta) {
            this.toast.error(error.response.data.meta.message);
          } else {
            this.toast.error(error.message);
          }
        }
      } else {
        this.form.role_id = this.selectedStatus.value;
        const dataToSend = {
          ...this.form,
          group: 'INTERNAL_DASHBOARD',
          is_active: 'TRUE',
        };
        const {
          data: { data },
        } = await API.post('/employees', dataToSend);
        this.handleAddAdmin(data.id, 'success create admin');
      }
    },
    async uploadS3(callback) {
      if (this.imageFile.file.size > 2000000) {
        this.toast.error('Oops, your image cannot be larger than 2MB');
      } else {
        this.loading = true;
        const S3Params = {
          Bucket: 'help-bns-bucket',
          Key: this.imageFile.fileName,
          Body: this.imageFile.file,
          ContentType: this.imageFile.type,
        };

        try {
          const S3Response = await this.s3.send(new PutObjectCommand(S3Params));
          if (S3Response) {
            callback(S3Response);
          }
        } catch (error) {
          this.toast.error(error.message);
        }
      }
    },
    handleAddAdmin(id, successMessage) {
      this.uploadS3(async (S3Response) => {
        const BNSParams = {
          title: this.form.name,
          hyperlink: this.hyperlink,
          bannerable: {
            id,
            type: 'EMPLOYEE',
          },
          group: 'COVER',
          provider: {
            name: 'S3',
            config: {
              location: this.imageFile.url,
              etag: S3Response.ETag.slice(1, -1),
              bucket: 'help-bns-bucket',
              key: this.imageFile.fileName,
            },
          },
        };

        try {
          this.loading = true;
          await API.post('/banners', BNSParams);
          this.$emit('reload');
          this.$emit('close');
          this.toast.success(successMessage);
        } catch (error) {
          this.toast.error(error.message);
        }
      });
      this.loading = false;
    },
    handleEditAdmin(id) {
      const payload = {
        title: this.form.name,
        hyperlink: this.hyperlink,
        bannerable: {
          id,
          type: 'EMPLOYEE',
        },
        group: 'COVER',
      };
      if (this.imageIsChanged) {
        this.uploadS3(async (S3Response) => {
          payload.provider = {
            name: 'S3',
            config: {
              location: this.imageFile.url,
              etag: S3Response.ETag.slice(1, -1),
              bucket: 'help-bns-bucket',
              key: this.imageFile.fileName,
            },
          };
          this.patchBNS(payload);
        });
      } else {
        this.patchBNS(payload);
      }
    },
    async patchBNS(payload) {
      try {
        this.loading = true;
        await API.patch(`banners/${this.bannerID}`, payload);
        this.$emit('reload');
        this.$emit('close');
        this.toast.success(`${this.form.name}'s profile has been successfully edited !`);
      } catch (error) {
        this.toast.error(error.message);
      }
      this.loading = false;
    },
  },
  mounted() {
    if (this.adminType === 'edit') {
      this.bannerID = this.$store.state.adminDetail.banner.id;
      this.form.name = this.$store.state.adminDetail.name || '';
      this.form.email = this.$store.state.adminDetail.email || '';
      this.form.phone_number = this.$store.state.adminDetail.phone_number || '';
      this.form.address = this.$store.state.adminDetail.address || '';
      this.form.role_id = this.$store.state.adminDetail.role_id || '';
      this.selectedStatus.label = this.$store.state.adminDetail.role || '';
      this.form.date_of_birth = this.$store.state.adminDetail.profile.date_of_birth || null;
      this.src = this.$store.state.adminDetail.banner
        ? this.$store.state.adminDetail.banner.location
        : '';
    }
  },
  computed: {
    screenWidth() {
      return this.$store.state.screenWidth;
    },
    s3() {
      return new S3Client({
        region: 'ap-southeast-1',
        credentials: fromCognitoIdentityPool({
          client: new CognitoIdentityClient({ region: 'ap-southeast-1' }),
          identityPoolId: process.env.VUE_APP_ID_POOL_ID,
        }),
      });
    },
  },
};
</script>

<style></style>
