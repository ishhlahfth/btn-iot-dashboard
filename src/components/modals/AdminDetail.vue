<template>
  <div class="grid grid-flow-rows gap-6 modal-md p-6">
    <div class="flex justify-center pb-6">
      <help-avatar :size="160" src="https://cdn.pixabay.com/photo/2017/07/19/17/22/test-external-2519783_1280.jpg"/>
    </div>
    <div class="grid grid-cols-2">
      <div>
        <p class="font-medium text-grey-2">Name</p>
        <p class="font-medium">Kiara</p>
      </div>
      <div>
        <p class="font-medium text-grey-2">Email</p>
        <p class="font-medium">Kiaradong@mail</p>
      </div>
    </div>
    <div class="grid grid-cols-2">
      <div>
        <p class="font-medium text-grey-2">Role</p>
        <p class="font-medium">Admin</p>
      </div>
      <div>
        <p class="font-medium text-grey-2">Phone Number</p>
        <p class="font-medium">081271062214</p>
      </div>
    </div>
    <div>
      <p class="font-medium text-grey-2">Address</p>
      <p class="font-medium">Lorem ipsum dolor sit amet consectetur adipisicing elit. Provident explicabo ut mollitia, omnis voluptatum repudiandae, fugit ipsum voluptates saepe esse quidem quasi dolores, blanditiis ullam neque id earum illo a.</p>
    </div>
  </div>
</template>

<script>
import HelpAvatar from '@/components/atoms/Avatar.vue';

export default {
  name: 'AdminDetail',
  components: {
    HelpAvatar,
  },
};
</script>

<style></style>
