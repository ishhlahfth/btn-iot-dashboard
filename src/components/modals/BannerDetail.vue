<template>
  <div
    class="grid gap-6 modal-lg inner-modal-auto overflow-auto"
    :class="[{ 'animate-pulse': loading }]"
  >
    <div class="flex justify-between items-center">
      <p class="text-heading4 font-semibold">Banner detail</p>
      <help-button
        icon-only
        icon="close"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
    </div>

    <div class="grid gap-2 md:gap-4 md:grid-flow-col md:grid-cols-12 mb-4">
      <div class="md:col-span-5 grid gap-2 md:gap-8 font-medium">
        <div class="grid gap-1">
          <p class="text-grey-2">Title</p>
          <p>{{ banner.title }}</p>
        </div>

        <div class="grid gap-1">
          <p class="text-grey-2">Redirect URL</p>
          <a
            :href="banner.hyperlink"
            target="blank"
            class="text-royal cursor-pointer hover:underline"
            >{{ banner.hyperlink }}</a
          >
        </div>

        <div class="grid grid-cols-2 gap-4">
          <div class="grid gap-1">
            <p class="text-grey-2">Starts at</p>
            <p>{{ banner.start_date }}</p>
          </div>
          <div class="grid gap-1">
            <p class="text-grey-2">Ends at</p>
            <p>{{ banner.end_date }}</p>
          </div>
        </div>
      </div>

      <div class="md:col-span-7 md:grid template-rows-auto-1fr-auto">
        <p class="font-medium mb-1 text-grey-2">Banner image</p>
        <help-thumbnail
          class="mb-1"
          width="100%"
          :src="banner.image_url"
          :height="screenWidth < 640 && !banner.image_url ? 128 : ''"
        >
          <div class="grid gap-2 place-items-center text-grey-2 p-4">
            <icon name="photograph" :size="6" />
            <p class="font-medium text-center">
              No image
            </p>
          </div>
        </help-thumbnail>
      </div>
    </div>
  </div>
</template>

<script>
import HelpButton from '@/components/atoms/Button.vue';
import HelpThumbnail from '@/components/atoms/Thumbnail.vue';
import Icon from '@/components/atoms/Icon.vue';
import { useToast } from 'vue-toastification';
import API from '@/apis';
import dayjs from 'dayjs';

export default {
  name: 'BannerDetail',
  components: {
    HelpButton,
    HelpThumbnail,
    Icon,
  },
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      banner: {},
      loading: false,
    };
  },
  computed: {
    screenWidth() {
      return this.$store.state.screenWidth;
    },
    bannerId() {
      return this.$store.state.bannerId;
    },
  },
  methods: {
    async getBanner() {
      try {
        const {
          data: { data },
        } = await API.get(`banners/${this.bannerId}`);

        if (data) {
          this.banner = {
            ...data,
            start_date: dayjs(data.start_date).format('ddd, D MMM YYYY'),
            end_date: dayjs(data.end_date).format('ddd, D MMM YYYY'),
          };
          const response = await this.$store.dispatch('loadImage', data.url);
          this.banner.image_url = response;
          console.log('IMG', response);
        }
      } catch (error) {
        this.toast.error(error.message);
      }
    },
  },
  mounted() {
    this.getBanner();
  },
};
</script>

<style lang="scss" scoped>
.template-rows-auto-1fr-auto {
  grid-template-rows: auto 1fr auto;
}
</style>
