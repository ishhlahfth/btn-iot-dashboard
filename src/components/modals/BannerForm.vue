<template>
  <div
    class="grid gap-6 inner-modal-auto modal-lg overflow-auto"
    :class="{ 'animate-pulse': loading }"
  >
    <div class="flex justify-between items-center">
      <p class="text-heading4 font-semibold">Add a new banner</p>
      <help-button
        icon-only
        icon="close"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
    </div>

    <div class="grid gap-4 md:grid-flow-col md:grid-cols-12 mb-4">
      <div class="md:col-span-5 grid gap-2">
        <help-input label="Title" placeholder="Give your banner a distinctive title" />
        <help-input label="Link to article" placeholder="http://www.redirect-here.wehelpyou.xyz" />
        <div class="grid md:grid-cols-2 gap-4">
          <help-input label="Starts at" placeholder="##-##-##" />
          <help-input label="Ends at" placeholder="##-##-##" />
        </div>
      </div>

      <div class="md:col-span-7">
        <p class="font-medium mb-1">Banner image</p>
        <help-thumbnail src="" width="100%" :height="screenWidth < 640 ? 128 : !src ? '100%' : ''">
          <help-button label="upload" class="mt-2" />
        </help-thumbnail>
      </div>
    </div>

    <div class="grid grid-flow-col gap-2 auto-cols-max justify-end">
      <help-button
        type="button"
        label="cancel"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
      <help-button label="save" />
    </div>
  </div>
</template>

<script>
import HelpButton from '@/components/atoms/Button.vue';
import HelpInput from '@/components/atoms/Input.vue';
import HelpThumbnail from '@/components/atoms/Thumbnail.vue';

export default {
  name: 'BannerForm',
  components: {
    HelpButton,
    HelpInput,
    HelpThumbnail,
  },
  data() {
    return {
      loading: false,
      src: '',
    };
  },
  computed: {
    screenWidth() {
      return this.$store.state.screenWidth;
    },
  },
};
</script>

<style></style>
