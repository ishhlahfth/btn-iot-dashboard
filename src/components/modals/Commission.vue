<template>
  <div class="grid gap-6 inner-modal-auto modal-md">
    <div class="flex justify-between items-center">
      <p class="text-heading4 font-semibold">Edit Commission</p>
      <help-button
        icon-only
        icon="close"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
    </div>
    <div class="grid sm:grid-cols-2 auto-rows-max gap-4 sm:gap-6 font-medium">
      <div>
        <p class="text-grey-2">Merchant Name</p>
        <p>{{ merchantName }}</p>
      </div>
      <div>
        <p class="text-grey-2">Current Commission</p>
        <p v-if="!loading">{{ commissionDetail.formula.value }} %</p>
        <div v-else class="rounded bg-grey-4 h-4 animate-pulse"></div>
      </div>
      <div class="sm:col-span-2">
        <form @submit.prevent="proceed" class="grid gap-4">
          <help-input
            label="New Commission"
            label-class="text-grey-2"
            v-model="newCommission"
            placeholder="Type new commission value here"
          />
          <help-button label="save changes" @click="updateCommission" />
        </form>
      </div>
    </div>
  </div>
</template>

<script>
import HelpButton from '@/components/atoms/Button.vue';
import HelpInput from '@/components/atoms/Input.vue';
import { useToast } from 'vue-toastification';
import API from '@/apis';

export default {
  name: 'Commission',
  components: {
    HelpButton,
    HelpInput,
  },
  emits: ['closeAndRefetch', 'close', 'refetch'],
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      commissionDetail: {
        formula: { value: 0 },
      },
      loading: false,
      newCommission: '',
    };
  },
  computed: {
    merchantId() {
      return this.$store.state.commissionDetail.merchantId;
    },
    merchantName() {
      return this.$store.state.commissionDetail.merchantName;
    },
  },
  methods: {
    async getCommission() {
      this.loading = true;
      try {
        const {
          data: { data },
        } = await API.get(`merchants/${this.merchantId}/bill-formulas`);
        if (data) {
          if (data.length) {
            this.commissionDetail = data.filter((el) => el.order_type_id === 4)[0];
          }
        }
      } catch (error) {
        console.log(error);
      }
      this.loading = false;
    },
    async updateCommission() {
      const payload = {
        merchant_id: this.commissionDetail.merchant_id,
        order_type_id: this.commissionDetail.order_type_id,
        seq_no: this.commissionDetail.seq_no,
        label: this.commissionDetail.label,
        formula: { type: 'PERCENT', value: +this.newCommission },
        formula_type: this.commissionDetail.formula_type,
      };
      this.loading = true;
      try {
        await API.patch(`bill-formulas/${this.commissionDetail.id}`, payload);

        this.getCommission();
        this.$emit('refetch');
        this.toast.success(`Successfully updated ${this.merchantName}`);
      } catch (error) {
        console.log(error);
      }
      this.loading = false;
    },
  },
  mounted() {
    this.getCommission();
  },
};
</script>

<style></style>
