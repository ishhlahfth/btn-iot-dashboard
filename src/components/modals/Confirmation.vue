<template>
  <div class="grid gap-6 inner-modal-auto modal-sm">
    <div class="flex justify-between items-center">
      <p class="text-heading4 font-semibold">{{ title }}</p>
      <help-button
        icon-only
        icon="close"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
    </div>

    <div class="grid auto-rows-max gap-4 sm:gap-6 font-medium text-grey-2">
      <p>{{ message }}</p>
    </div>

    <div class="flex flex-row justify-end">
      <help-button
        @click="$emit('cancel')"
        label="cancel"
        class="mb-2 sm:mb-0 sm:mr-2"
        bg-color="transparent"
        color="grey-1"
      />
      <help-button
        :bg-color="bgColor"
        @click="$emit('confirm')"
        label="yes i'm sure"
        class="mb-2 sm:mb-0"
        :loading="confirmLoading"
        :loading-label="loadingLabel"
      />
    </div>
  </div>
</template>

<script>
import HelpButton from '@/components/atoms/Button.vue';

export default {
  name: 'Confirmation',
  props: {
    title: {
      type: String,
      default: '',
    },
    message: {
      type: String,
      default: '',
    },
    confirmLoading: {
      type: Boolean,
      default: false,
    },
    loadingLabel: {
      type: String,
      default: '',
    },
    bgColor: {
      type: String,
      default: 'midnight',
    },
  },
  components: {
    HelpButton,
  },
};
</script>

<style></style>
