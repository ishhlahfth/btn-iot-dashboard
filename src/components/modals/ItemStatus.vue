<template>
  <div class="grid gap-6 inner-modal-auto modal-sm">
    <div class="flex justify-between items-center">
      <p class="text-heading4 font-semibold">Edit Item Status</p>
      <help-button
        icon-only
        icon="close"
        bg-color="transparent"
        color="grey-1"
        @click="$emit('close')"
      />
    </div>
    <form @submit.prevent="proceed" class="grid gap-4">
      <div class="w-full">
        <help-select
          label="Change status to"
          :options="statuses"
          v-model="selectedStatus"
          :position="screenWidth < 640 ? ['top', 'right'] : ['bottom', 'right']"
        />
      </div>
      <help-button label="proceed" />
    </form>
  </div>
</template>

<script>
import { useToast } from 'vue-toastification';
import HelpButton from '@/components/atoms/Button.vue';
import HelpSelect from '@/components/molecules/Select.vue';
import API from '../../apis';

export default {
  name: 'MerchantVerification',
  components: {
    HelpButton,
    HelpSelect,
  },
  emits: ['close'],
  setup() {
    const toast = useToast();
    return { toast };
  },
  data() {
    return {
      statuses: [
        { value: 'AVAILABLE', label: 'Tersedia' },
        { value: 'UNAVAILABLE', label: 'Habis' },
        // { value: 'HIDDEN', label: 'Hidden' },
        // { value: 'OUT_OF_STOCK', label: 'Out of Stock' },
        { value: 'SUSPEND', label: 'Suspend' },
      ],
      selectedStatus: { value: 'AVAILABLE', label: 'Tersedia' },
    };
  },
  computed: {
    item() {
      return this.$store.state.item;
    },
    screenWidth() {
      return this.$store.state.screenWidth;
    },
  },
  methods: {
    async proceed() {
      const payload = {
        status: this.selectedStatus.value, // 'AVAILABLE', 'UNAVAILABLE', 'HIDDEN', 'OUT_OF_STOCK', 'SUSPEND'
        sort_no: this.item.sort_no,
        name: this.item.name,
        description: this.item.description,
        price: this.item.price,
        min_buy_qty: this.item.min_buy_qty,
        catalog_id: this.item.catalog.id,
        group_id: this.item.group.id,
        variations: this.item.variations.map((variation) => ({
          ...variation,
          id: variation.variation_id,
          options: variation.options.map((option) => ({
            ...option,
            id: option.option_id,
          })),
        })),
      };

      try {
        const {
          data: { data },
        } = await API.patch(`items/${this.item.id}`, payload);

        this.toast.success(`Successfully updated ${data.name}`);
        this.$store.dispatch('loadMerchant', this.$store.state.merchantId);
        this.$emit('close');
      } catch (error) {
        console.log(error);
      }
    },
  },
};
</script>

<style></style>
