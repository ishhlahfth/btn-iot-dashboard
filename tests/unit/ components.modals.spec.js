import AdminAddEdit from '@/components/modals/AdminAddEdit.vue';
import AdminDetail from '@/components/modals/AdminDetail.vue';
import BannerDetail from '@/components/modals/BannerDetail.vue';
import BannerForm from '@/components/modals/BannerForm.vue';
import Commission from '@/components/modals/Commission.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
import ItemStatus from '@/components/modals/ItemStatus.vue';
import ListAdmin from '@/components/modals/ListAdmin.vue';
import MerchantDetail from '@/components/modals/MerchantDetail.vue';
import MerchantFilter from '@/components/modals/MerchantFilter.vue';
import MerchantVerification from '@/components/modals/MerchantVerification.vue';
import MerchantVerificationOption from '@/components/modals/MerchantVerificationOption.vue';
import OperationalHour from '@/components/modals/OperationalHour.vue';
import OrderDetail from '@/components/modals/OrderDetail.vue';
import OrderFilter from '@/components/modals/OrderFilter.vue';
import RoleAdd from '@/components/modals/RoleAdd.vue';
import RoleDetail from '@/components/modals/RoleDetail.vue';
import StatusHistory from '@/components/modals/StatusHistory.vue';
import TransferFilter from '@/components/modals/TransferFilter.vue';
import { shallowMount } from '@vue/test-utils';

describe('AdminAddEdit.vue', () => {
  it('should render correctly', () => {
    expect(AdminAddEdit.$el).toMatchSnapshot();
  });

  it('should render name AdminAddEdit', () => {
    expect(AdminAddEdit.name).toBe('AdminAddEdit');
  });
});

describe('AdminDetail.vue', () => {
  it('should render correctly', () => {
    expect(AdminDetail.$el).toMatchSnapshot();
  });

  it('should render name AdminDetail', () => {
    expect(AdminDetail.name).toBe('AdminDetail');
  });
});

describe('BannerDetail.vue', () => {
  it('should render correctly', () => {
    expect(BannerDetail.$el).toMatchSnapshot();
  });

  it('should render name BannerDetail', () => {
    expect(BannerDetail.name).toBe('BannerDetail');
  });
});

describe('BannerForm.vue', () => {
  it('should render correctly', () => {
    expect(BannerForm.$el).toMatchSnapshot();
  });

  it('should render name BannerForm', () => {
    expect(BannerForm.name).toBe('BannerForm');
  });
});

describe('Commission.vue', () => {
  it('should render correctly', () => {
    expect(Commission.$el).toMatchSnapshot();
  });

  it('should render name Commission', () => {
    expect(Commission.name).toBe('Commission');
  });
});

describe('Confirmation.vue', () => {
  it('should render correctly', () => {
    expect(Confirmation.$el).toMatchSnapshot();
  });

  it('should render name Confirmation', () => {
    expect(Confirmation.name).toBe('Confirmation');
  });
});

describe('ItemStatus.vue', () => {
  it('should render correctly', () => {
    expect(ItemStatus.$el).toMatchSnapshot();
  });

  it('should render name ItemStatus', () => {
    expect(ItemStatus.name).toBe('ItemStatus');
  });
});

describe('ListAdmin.vue', () => {
  it('should render correctly', () => {
    expect(ListAdmin.$el).toMatchSnapshot();
  });

  it('should render name ListAdmin', () => {
    expect(ListAdmin.name).toBe('ListAdmin');
  });
});

describe('MerchantDetail.vue', () => {
  it('should render correctly', () => {
    expect(MerchantDetail.$el).toMatchSnapshot();
  });

  it('should render name MerchantDetail', () => {
    expect(MerchantDetail.name).toBe('MerchantDetail');
  });
});

describe('MerchantFilter.vue', () => {
  it('should render correctly', () => {
    expect(MerchantFilter.$el).toMatchSnapshot();
  });

  it('should render name MerchantFilter', () => {
    expect(MerchantFilter.name).toBe('MerchantFilter');
  });
});

describe('MerchantVerification.vue', () => {
  it('should render correctly', () => {
    expect(MerchantVerification.$el).toMatchSnapshot();
  });

  it('should render name MerchantVerification', () => {
    expect(MerchantVerification.name).toBe('MerchantVerification');
  });
});

describe('MerchantVerificationOption.vue', () => {
  it('should render correctly', () => {
    expect(MerchantVerificationOption.$el).toMatchSnapshot();
  });

  it('should render name MerchantVerificationOption', () => {
    expect(MerchantVerificationOption.name).toBe('MerchantVerificationOption');
  });
});

describe('OperationalHour.vue', () => {
  it('should render correctly', () => {
    expect(OperationalHour.$el).toMatchSnapshot();
  });

  it('should render name OperationalHour', () => {
    expect(OperationalHour.name).toBe('OperationalHour');
  });
});

describe('OrderDetail.vue', () => {
  it('should render correctly', () => {
    expect(OrderDetail.$el).toMatchSnapshot();
  });

  it('should render name OrderDetail', () => {
    expect(OrderDetail.name).toBe('OrderDetail');
  });
});

describe('OrderFilter.vue', () => {
  it('should render correctly', () => {
    expect(OrderFilter.$el).toMatchSnapshot();
  });

  it('should render name OrderFilter', () => {
    expect(OrderFilter.name).toBe('OrderFilter');
  });
});

describe('RoleAdd.vue', () => {
  it('should render correctly', () => {
    expect(RoleAdd.$el).toMatchSnapshot();
  });

  it('should render name RoleAdd', () => {
    expect(RoleAdd.name).toBe('RoleAdd');
  });
});

describe('RoleDetail.vue', () => {
  it('should render correctly', () => {
    expect(RoleDetail.$el).toMatchSnapshot();
  });

  it('should render name RoleDetail', () => {
    expect(RoleDetail.name).toBe('RoleDetail');
  });
});

describe('StatusHistory.vue', () => {
  it('should render correctly', () => {
    expect(StatusHistory.$el).toMatchSnapshot();
  });

  it('should render name StatusHistory', () => {
    expect(StatusHistory.name).toBe('StatusHistory');
  });
});

describe('TransferFilter.vue', () => {
  it('should render correctly', () => {
    expect(TransferFilter.$el).toMatchSnapshot();
  });

  it('should render name TransferFilter', () => {
    expect(TransferFilter.name).toBe('TransferFilter');
  });
});
