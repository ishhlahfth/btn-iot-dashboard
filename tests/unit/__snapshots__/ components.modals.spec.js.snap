// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminAddEdit.vue should render correctly 1`] = `undefined`;

exports[`AdminDetail.vue should render correctly 1`] = `undefined`;

exports[`BannerDetail.vue should render correctly 1`] = `undefined`;

exports[`BannerForm.vue should render correctly 1`] = `undefined`;

exports[`Commission.vue should render correctly 1`] = `undefined`;

exports[`Confirmation.vue should render correctly 1`] = `undefined`;

exports[`ItemStatus.vue should render correctly 1`] = `undefined`;

exports[`ListAdmin.vue should render correctly 1`] = `undefined`;

exports[`MerchantDetail.vue should render correctly 1`] = `undefined`;

exports[`MerchantFilter.vue should render correctly 1`] = `undefined`;

exports[`MerchantVerification.vue should render correctly 1`] = `undefined`;

exports[`MerchantVerificationOption.vue should render correctly 1`] = `undefined`;

exports[`OperationalHour.vue should render correctly 1`] = `undefined`;

exports[`OrderDetail.vue should render correctly 1`] = `undefined`;

exports[`OrderFilter.vue should render correctly 1`] = `undefined`;

exports[`RoleAdd.vue should render correctly 1`] = `undefined`;

exports[`RoleDetail.vue should render correctly 1`] = `undefined`;

exports[`StatusHistory.vue should render correctly 1`] = `undefined`;

exports[`TransferFilter.vue should render correctly 1`] = `undefined`;
